scripts/gen-upgrade-proposal.sh

```diff
@@ -1,43 +1,45 @@
 #! /bin/bash
 set -ueo pipefail
 
-cat << EOF 1>&2
-------------------------------------------------------------
+cat <<'EOF' 1>&2
+------------------------------------------------------------------------
 This script shows a command for a software upgrade proposal which is
 compatible with the Cosmovisor found at:
 https://github.com/agoric-labs/cosmos-sdk/tree/Agoric/cosmovisor#readme
-------------------------------------------------------------
+------------------------------------------------------------------------
 EOF
 
 COMMIT_ID=$(git rev-parse HEAD)
-ZIPURL=https://github.com/Agoric/agoric-sdk/archive/$COMMIT_ID.zip
+ZIP_URL="https://github.com/Agoric/agoric-sdk/archive/${COMMIT_ID}.zip"
 
-echo "Verifying archive is at $ZIPURL..." 1>&2
+echo "Verifying archive is at $ZIP_URL..." 1>&2
 zipfile=$(mktemp)
 trap 'rm -f "$zipfile"' EXIT
-curl -L "$ZIPURL" -o "$zipfile"
+curl -L "$ZIP_URL" -o "$zipfile"
 
 echo "Generating SHA-256 checksum..." 1>&2
-checksum=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
+CHECKSUM=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
```

Comment:
The code in forum posts is not assumed to be running inside a script, and so doesn't have this download-with-deferred-cleanup behavior. But unfortunately, that means it also doesn't have verification that the archive exists. We could get that by adding a HEAD request both here and in future posts, e.g.
```sh
echo "Verifying archive is at $ZIP_URL..." 1>&2
curl -sSLI "$ZIP_URL" | grep -E '^HTTP/[0-9.]+ 200\b'

echo "Generating SHA-256 checksum..." 1>&2
CHECKSUM=sha256:$(curl -L "$ZIP_URL" -o- | shasum -a 256 | cut -d' ' -f1)
```

Reply:
I'd prefer using the `-f` flag to have `curl` fail fast:

```sh
echo "Verifying ${ZIP_URL} presence and generating SHA-256 checksum..." 1>&2
CHECKSUM=sha256:$(curl -fL "$ZIP_URL" -o- | shasum -a 256 | cut -d' ' -f1)
```

Reply:
As noted above, the code in forum posts is not assumed to be running inside a script—so `pipefail` may not be active or even available (our instructions don't require bash). For example, here's what I see when running in a clean shell:
```console
$ CHECKSUM=sha256:$(curl -L 'https://example.invalid/' -o- | shasum -a 256 | cut -d' ' -f1)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (6) Could not resolve host: example.invalid
$ echo "CHECKSUM: $CHECKSUM"
CHECKSUM: sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

But I do like the idea of using `-f`, and have updated accordingly.